bcrypt/auth.go

```
@@ -0,0 +1,33 @@
+package bcrypt
+
+import (
+	"log"
+	"golang.org/x/crypto/bcrypt"
+)
+
+type UserAuthenticator struct {
+}
+
+func (a *UserAuthenticator)HashAndSalt(pwd string) string {
```

It's better to return the error up to the caller of the function and let them handle the failure. This way, the returned hash would be just `""` without anyone noticing.